### PR TITLE
td/fix rt for ascend

### PR DESCRIPTION
**Pull request by chrysantd**
## Motivation and Context
old ascend DIOPI runtime have several bugs under current DIOPI version, which causes

## Description
1. add initialization for acl library
2. release python gil when entering cpp code of operators in pybind level
3. fix some headers included
4. improve debug macro of CALL_ACLRT

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: DIOPI-IMPL/ascend/functions/error.cpp [hunk 1]
```diff
@@ -3,6 +3,8 @@
 #include <acl/acl.h>
 #include <diopi/functions.h>
 
+#include <cstdio>
```
Comment (xintian-514) [inline DIOPI-IMPL/ascend/functions/error.cpp:6]: 这个是为什么？

File: DIOPI-IMPL/ascend/test/conform_test.cpp [hunk 2]
```diff
@@ -62,6 +54,14 @@ int32_t device_memcpy_d2d_async(diopiStreamHandle_t stream_handle, void* dst, co
     return diopiSuccess;
 }
 
+int32_t initLibrary() {
+    CALL_ACLRT(aclInit(NULL));
+    CALL_ACLRT(aclrtSetDevice(0));
+    aclrtContext context_;
```
Comment (xintian-514) [inline DIOPI-IMPL/ascend/test/conform_test.cpp:60]: 这个context_的作用是？

File: DIOPI-TEST/codegen/op_template.py [hunk 3]
```diff
@@ -88,6 +88,7 @@ class OpTemplate(object):
     function_template = CodeTemplate("""\
 m.def("${func_name}", [](${attrs}) {
     if (${func_name}) {
+        py::gil_scoped_release no_gil;
```
Comment (xintian-514) [inline DIOPI-TEST/codegen/op_template.py:91]: 这个的作用是？

File: DIOPI-IMPL/ascend/test/conform_test.cpp (hunk 2 above)
Comment (chrysantd) [reply DIOPI-IMPL/ascend/test/conform_test.cpp:60]: aclrtCreateContext下面aclrtCreateContext不是要用吗

File: DIOPI-TEST/codegen/op_template.py (hunk 3 above)
Comment (chrysantd) [reply DIOPI-TEST/codegen/op_template.py:91]: 华为执行算子时会使用多线程，如果不release gil的话会出错

File: DIOPI-IMPL/ascend/functions/error.cpp (hunk 1 above)
Comment (chrysantd) [reply DIOPI-IMPL/ascend/functions/error.cpp:6]: sprintf需要

**Pull request closed without merging**